src/validations/index.js

```diff
@@ -1 +1,61 @@
-// this folder will contain validation middlewares' definitions.
\ No newline at end of file
+// this folder will contain validation middlewares' definitions.
+import Joi from 'joi';
+
+const schema = {
+	signupvalidate: Joi.object().keys({
+		email: Joi.string().required().email(),
+		firstName: Joi.string()
+			.empty()
+			.min(2)
+			.max(20)
+			.pattern(/^[a-zA-Z]/)
+			.messages({
+				'any.required': '{{#label}} field is required',
+				'string.base': '{{#label}} must be of type string',
+				'string.empty': '{{#label}} can not be empty',
+				'string.pattern.base':
+					'{{#label}} must contain only characters from a to z.',
+			}),
+		lastName: Joi.string()
+			.empty()
+			.min(2)
+			.max(20)
+			.pattern(/^[a-zA-Z]/)
+			.messages({
+				'any.required': '{{#label}} field is required',
+				'string.base': '{{#label}} must be of type string',
+				'string.empty': '{{#label}} can not be empty',
+				'string.pattern.base':
+					'{{#label}} must contain only characters from a to z.',
+			}),
+		password: Joi.string()
+			.required()
+			.empty()
+			.pattern(/^(?=.*[A-Z])(?=.*[0-9])(?=.*[@#*&]+)[\w@#*&]{8,}$/)
+			.messages({
+				'any.required': '{{#label}} field is required',
+				'string.base': '{{#label}} must be of type string',
+				'string.empty': '{{#label}} can not be empty',
+				'string.pattern.base':
+					'{{#label}} must contain at least a number, a special character, an upper-case letter and longer than 8 characters',
+			}),
+		role: Joi.string(),
+	}),
+};
+
+const { signupvalidate } = schema;
+
+class AuthValidation {
+	static async verifySignup(req, res, next) {
+		const { error } = signupvalidate.validate(req.body);
+		if (error) {
+			throw new Error(
```

Comment:
I have tested (manually) this feature, and I found that on validation, this line throws an error when bad credentials are sent (for registration). 
The problem that I found is that such `throw new Error()` scripts causes the server to crash, and end-user applications (for e.g postman) can re-gain access to the server only if we re-run `npm run dev` from the terminal.
Therefore, I would suggest that line 51 to 58 may be changed to something like:
```javascript
if (error) {
    res.status(400).send({
        error: error.details[0].message.replace(/["'`]+/g, ''),
    })
}
else{
    return next();
}
```